source/SharpGL/Core/SharpGL.WPF/OpenGLControl.xaml.cs

```diff
@@ -31,21 +23,87 @@ public partial class OpenGLControl : UserControl
         public OpenGLControl()
         {
             InitializeComponent();
-            SizeChanged += new SizeChangedEventHandler(OpenGLControl_SizeChanged);
+
+            Unloaded += OpenGLControl_Unloaded;
+            Loaded += OpenGLControl_Loaded;
+
+            var frameworkElement = Parent as FrameworkElement;
+
+            if (frameworkElement != null)
+            {
+                frameworkElement.Loaded += Parent_Loaded;
+                frameworkElement.Unloaded += Parent_Unloaded;
+            }
+        }
+
+        /// <summary>
+        /// Handles the Loaded event of the parent control.
+        /// </summary>
+        /// <param name="sender">The source of the event.</param>
+        /// <param name="e">The <see cref="System.Windows.RoutedEventArgs"/> Instance containing the event data.</param>
+        private void Parent_Loaded(object sender, RoutedEventArgs e)
+        {
+            if (timer != null)
+            {
+                timer.Tick -= timer_Tick;
+                timer.Tick += timer_Tick; // to avoid twice call, ensure timer_Tick event handler is registered only once, because it is also registred on apply template
+            }
+        }
+
+        /// <summary>
+        /// Handles the Unloaded event of the parent control.
+        /// </summary>
+        /// <param name="sender">The source of the event.</param>
+        /// <param name="e">The <see cref="System.Windows.RoutedEventArgs"/> Instance containing the event data.</param>
+        private void Parent_Unloaded(object sender, RoutedEventArgs e)
+        {
+            if (timer != null)
+                timer.Tick -= timer_Tick;
+        }
+
+        /// <summary>
+        /// Handles the Loaded event of the OpenGLControl control.
+        /// </summary>
+        /// <param name="sender">The source of the event.</param>
+        /// <param name="e">The <see cref="System.Windows.RoutedEventArgs"/> Instance containing the event data.</param>
+        private void OpenGLControl_Loaded(object sender, RoutedEventArgs routedEventArgs)
+        {
+            SizeChanged += OpenGLControl_SizeChanged;
+
+            UpdateOpenGLControl((int) RenderSize.Width, (int) RenderSize.Height);

```

Comment:
So we've wired up the size handler and resized, I think it's here we should now attach the timer event handler and start the timer, rather than in `OnApplyTemplate`